include/boost/geometry/algorithms/detail/envelope/linestring.hpp

```diff
@@ -144,19 +154,27 @@ struct envelope_linear_on_spheroid
                 typename coordinate_system<Box>::type::units
             >::apply(longitude_intervals, lon_min, lon_max);
 
-        assign_values(mbr, lon_min, lat_min, lon_max, lat_max);
+        geometry::set<min_corner, 0>(mbr, lon_min);
+        geometry::set<min_corner, 1>(mbr, lat_min);
+        geometry::set<max_corner, 0>(mbr, lon_max);
+        geometry::set<max_corner, 1>(mbr, lat_max);
+
+        dispatch::envelope
+            <
+                Linear, 2, DimensionCount
+            >::apply(linear, mbr);
```

Comment:
I forgot about the most important thing. Please treat my comments about the possible optimizations and improvements as general remarks which may be postponed, since those things are not parts of this PR.